tracer/src/Datadog.Trace/Activity/OtlpHelpers.cs

```diff
@@ -0,0 +1,410 @@
+                {
+                    if (span.GetTag("otel.status_description") is string statusDescription)
+                    {
+                        span.SetTag(Tags.ErrorMsg, statusDescription);
+                    }
+                    else if (span.GetTag("http.status_code") is string statusCodeString)
+                    {
+                        if (span.GetTag("http.status_text") is string httpTextString)
+                        {
+                            span.SetTag(Tags.ErrorMsg, $"{statusCodeString} {httpTextString}");
+                        }
+                        else
+                        {
+                            span.SetTag(Tags.ErrorMsg, statusCodeString);
+                        }
+                    }
+                }
+            }
+        }
+
+        // See trace agent func spanKind2Type: https://github.com/DataDog/datadog-agent/blob/67c353cff1a6a275d7ce40059aad30fc6a3a0bc1/pkg/trace/api/otlp.go#L621
+        internal static string AgentSpanKind2Type(ActivityKind kind, Span span) => kind switch
+        {
+            ActivityKind.Server => SpanTypes.Web,
+            ActivityKind.Client => span.GetTag("db.system") switch
+            {
+                "redis" or "memcached" => "cache",
+                string => "db",
+                _ => "http",
+            },
+            _ => SpanTypes.Custom,
+        };
+
+        internal static string ComposeArrayString(Array array)
+        {
+            if (array.Length == 0)
+            {
+                return "[]";
+            }
+
+            StringBuilder sb = StringBuilderCache.Acquire(0);
+            sb.Append('[');
+            for (int i = 0; i < array.Length; i++)
+            {
+                var value = array.GetValue(i);
+                if (value is null)
+                {
+                    sb.Append("null");
+                }
+                else if (value is string s)
+                {
+                    sb.Append('"');
+                    sb.Append(s);
+                    sb.Append('"');
+                }
+                else if (value is bool b)
+                {
+                    sb.Append(b ? "true" : "false");
+                }
+                else
+                {
+                    sb.Append(value.ToString());
+                }
+
+                sb.Append(',');
+            }
+
+            sb.Remove(sb.Length - 1, 1);
+            sb.Append(']');
+            return StringBuilderCache.GetStringAndRelease(sb);
+        }
+
+        // This algorithm implements Go function strconv.ParseBool, which the trace agent uses
+        // for string->bool conversions
+        // See https://pkg.go.dev/strconv#ParseBool
+        private static bool? GoStrConvParseBool(string value)
+        {
+            if (value == null) { ThrowHelper.ThrowArgumentNullException(nameof(value)); }
+
+            if (value.Length == 0)
+            {
+                return null;
+            }
```

Comment:
Should we handle null values and empty strings the same way, by returning null for both?

Reply:
Hmm, yeah I think you're right. Updated in b215b417d23964c2a89d55c83e1004c81b0c3642

Reply:
Thanks!